### PR TITLE
[bugfix] libgd: INT_MAX not defined

### PR DESCRIPTION
**Pull request by lang-m**
Compiling version 2.2.4 fails (on a Debian system with only a minimum set of packages installed) with an error because `INT_MAX` is undeclared:

```
   263    gd_gd2.c: In function '_gd2GetHeader':
>> 264    gd_gd2.c:212:54: error: 'INT_MAX' undeclared (first use in this function)
   265      212 |                 if (*ncx <= 0 || *ncy <= 0 || *ncx > INT_MAX / *ncy) {
   266          |                                                      ^~~~~~~
   267    gd_gd2.c:87:1: note: 'INT_MAX' is defined in header '<limits.h>'; did you forget to '#include <limits.h>'?
```

The patch is taken from upstream: https://github.com/libgd/libgd/commit/c9b601a658a79e6ea2aad29fbf60ca6e24ccef1e

### CODE REVIEW

**Review by bernhardkaindl — DISMISSED** [review]
LGTM

File: var/spack/repos/builtin/packages/libgd/package.py [hunk 1]
```diff
@@ -39,6 +39,8 @@ class Libgd(AutotoolsPackage):
     depends_on("fontconfig")
     depends_on("libx11")
 
+    patch("int_max.patch", when="@2.2.4")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/libgd/package.py]: Can we retrieve the patch from a url? Also, can you leave a brief comment above this directive to explain what this patch does?

Comment (lang-m) [reply var/spack/repos/builtin/packages/libgd/package.py]: Done

**Comment by alalazo** [comment]
Can you fix style? We can't do it with our bot because the PR comes from an organization (and you cannot let non-members edit the branch as far as I know).

File: var/spack/repos/builtin/packages/libgd/package.py [hunk 2]
```diff
@@ -39,6 +39,13 @@ class Libgd(AutotoolsPackage):
     depends_on("fontconfig")
     depends_on("libx11")
 
+    # add missing '#include <limits.h>' in gd_gd2.c, which uses the constant 'INT_MAX'
+    patch(
+        "https://github.com/libgd/libgd/commit/c9b601a658a79e6ea2aad29fbf60ca6e24ccef1e.patch?full_index=1",
+        sha256="6ad03ef2e2c5940edf14ba3c145b0a6e8201a2e83d65a68fe7c903c9263852fc",
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/libgd/package.py]: Can you double check the shasum? Here's what I get:
```
$ wget https://github.com/libgd/libgd/commit/c9b601a658a79e6ea2aad29fbf60ca6e24ccef1e.patch?full_index=1
$ sha256sum c9b601a658a79e6ea2aad29fbf60ca6e24ccef1e.patch\?full_index\=1 
1dc3a72491427acbae2cd0c6d3b08c0814ffa2f9fee91269b8b46429cabb773d  c9b601a658a79e6ea2aad29fbf60ca6e24ccef1e.patch?full_index=1

```

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/libgd/package.py (hunk 2 above)
Comment (lang-m) [reply var/spack/repos/builtin/packages/libgd/package.py]: You are right, the checksum is not correct. (It was the old one for the missing `?full_index=1`)

**Review by alalazo — APPROVED** [review]

**Pull request merged**